tests/test_read_into_dict.py

```diff
@@ -0,0 +1,41 @@
+"""Tests for the csvy reader functions."""
+
+import logging
+import warnings
+
+import pytest
+
+from csvy.readers import read_into_dict
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+data_files = [
+    "tests/data_comment.csv",
+    "tests/array_data.csv",
+    "tests/data.csv",
+    "tests/data_read_into_dict.csvy",
+]
+
+
+@pytest.mark.parametrize("file_path", data_files)
+def test_read_into_dict(file_path):
+    """Test the read_into_dict function with multiple datasets."""
+    data, metadata = read_into_dict(file_path)
```

Comment:
These values are never checked properly. The content of the `data` and `metadata` variables should be checked in a similar way to how they are checked in the other tests in `test_read.py`